pages/content/amp-dev/documentation/guides-and-tutorials/develop/custom-javascript-tutorial.md

```diff
@@ -0,0 +1,283 @@
+const lengthCheck = document.getElementById("eight");
+const submitButton = document.getElementById("submitButton");
+```
+
+
+Like non-AMP pages, we can test if they were captured correctly by logging elements into the console. 
+
+
+## Set checks
+
+The script file will hold all of our password requirement logic by using [regular expressions](https://en.wikipedia.org/wiki/Regular_expression) to check that passwords pass validation requirements. We want to communicate to our users when each check has been met, so each requirement has its own RegEx. Declare the checks object by adding the code below to `script.js`:
+
+
+```js
+const passwordChecks = [
+  {
+    checkRegEx: (password) => (password.search(/[a-z]/) >= 0),
+    text: lowerCheck
+  },
+  {
+    checkRegEx: (password) => (password.search(/[A-Z]/) >= 0),
+    text: upperCheck
+  },
+  {
+    checkRegEx: (password) => (password.search(/[0-9]/) >= 0),
+    text: numberCheck
+  },
+  {
+    checkRegEx: (password) => (password.search(/[^a-z0-9]/i) >= 0),
+    text: specialCheck
+  },
+  {
+    checkRegEx: (password) => (password.length >= 8),
+    text: lengthCheck
+  }
+]
+
+```
+
+
+
+## Create functionality
+
+Here is where we add our functionality, by creating a function! We'll declare one called `initCheckPassword` that takes a single argument. The argument will be the value our user enters into the password input. We’ll run it through our checks above and change our defined password rules text green or red, depending on if it has been met or not. 
+
+
+```js
+function initCheckPassword(element) {
+
+};
+```
+
+
+We’ll then see if the element we pass as our argument can pass the defined checks.
+
+
+```js
+function initCheckPassword(element) {
+ const checkPassword = () => {
+    passwordChecks.forEach((item) => {
+      let passed = item.checkRegEx(element.value);
+       // passed logic 
+    });
+};
+```
+
+User actions can trigger [mutations within the `amp-script` component ]({{g.doc('/content/amp-dev/documentation/components/amp-script.md', locale=doc.locale).url.path}}?format=websites#mutations-and-user-actions), allowing `amp-script` to register [event listeners](https://developer.mozilla.org/en-US/docs/Web/API/EventListener). 
+
+Our function will listen for two events, [`keyup`](https://developer.mozilla.org/en-US/docs/Web/API/Document/keyup_event) for when a user types into the input box, and [`change`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/change_event) in case our user pastes their password.
+
+
+```js
+function initCheckPassword(element) {
+ const checkPassword = () => {
+    passwordChecks.forEach((item) => {
+      let passed = item.checkRegEx(element.value);
+       // passed logic 
+    });
+}; 
+// is called when user types in input
+element.addEventListener("keyup", checkPassword);
+// is called if user pastes into input 
+element.addEventListener("change", checkPassword);  
+```
+
+
+Inside the `checkPassword` function, we’ll call two more:
```

Comment:
These section here could be simplified by changing the CSS. Instead of using two classes to encode the state, you could change the CSS to default to the fail state:

```
<style>
    .check {
      color:#c11136;
    }
    .check.pass {
      color: #2d7b1f;
    } 
  </style>

 ...

<li id="number" class="check">Number</li>
...
```

Then you can simplify the JS code to:

```
item.text.classList.toggle('pass', passed)
```

and you no longer need two extra functions.